File: qwen2.5-coder/sistemaTipos/tipos.py
def calcular_precio_final(precio_original, descuento):
    # Convertimos el descuento a un número flotante
    descuento = float(descuento)
    
    # Calculamos el monto del descuento
    monto_descuento = (descuento / 100) * precio_original
    
    # Restamos el monto del descuento al precio original para obtener el precio final
    precio_final = precio_original - monto_descuento
    
    return precio_final

# Llamada a la función con un precio de 50 y un descuento del "10%"
precio = 50
descuento = "10"
precio_final = calcular_precio_final(precio, descuento)

print(f"El precio final es: {precio_final}")

File: qwen2.5-coder/sistemaTipos/tipos.rs
use std::str::FromStr;

fn calcular_precio_final(precio: f64, descuento: &str) -> Option<f64> {
    // Intenta parsear el descuento a un número flotante
    match f64::from_str(descuento) {
        Ok(d) => {
            // Asegúrate de que el descuento esté entre 0 y 1 (0% - 100%)
            if d >= 0.0 && d <= 1.0 {
                let descuento_amount = precio * d;
                Some(precio - descuento_amount)
            } else {
                None // Descuento no válido
            }
        },
        Err(_) => None, // No se pudo convertir la cadena a número flotante
    }
}

fn main() {
    // Llamada a la función con precio 50 y descuento "10%"
    let precio_final = calcular_precio_final(50.0, "10");
    
    match precio_final {
        Some(precio) => println!("El precio final es: {}", precio),
        None => println!("Descuento no válido"),
    }
}